### PR TITLE
[R] add reset_data in lgb.cv

### LINKED ISSUE #1854: Please add reset_data to lgb.cv
The lgb.train has the argument `reset_data = T` to release the data from the model as in https://github.com/Microsoft/LightGBM/issues/879. In the case of multiple k-fold cv the RAM is eaten so fast, so it would much better if lgb.cv also has the argument `reset_data`. 

Thank you very much.

### PR DESCRIPTION
**Pull request by guolinke**
fix #1854

### CODE REVIEW

File: R-package/R/lgb.cv.R [hunk 1]
```diff
@@ -303,7 +305,9 @@ lgb.cv <- function(params = list(),
     if (env$met_early_stop) break
 
   }
-
+  if (free_boosters) {
+    cv_booster$boosters = NULL
```
[the remark this answers is not in the document]
Comment (jameslamb) [reply R-package/R/lgb.cv.R]: @guolinke would anyone ever NOT want to free them? I'd rather just make this new behavior and leave the function signature the same. Adding flags to the code makes it harder to maintain, in my experience.

**Comment by guolinke** [comment]
@jameslamb @Laurae2 
please review this PR. Thanks !

File: R-package/R/lgb.train.R [hunk 2]
```diff
@@ -12,6 +12,8 @@
 #' @param categorical_feature list of str or int
 #'        type int represents index,
 #'        type str represents feature names
+#' @param callbacks list of callback functions
+#'        List of callback functions that are applied at each iteration.
```
Comment (jameslamb) [inline R-package/R/lgb.train.R]: how is this change related?

File: R-package/R/lgb.cv.R [hunk 3]
```diff
@@ -303,7 +305,19 @@ lgb.cv <- function(params = list(),
     if (env$met_early_stop) break
 
   }
-
+  if (reset_data) {
+    lapply(cv_booster$boosters, function(fd) {
+      # Store temporarily model data elsewhere
```
Comment (jameslamb) [inline R-package/R/lgb.cv.R]: Makes sense to me

File: R-package/R/lgb.train.R (hunk 2 above)
Comment (guolinke) [reply R-package/R/lgb.train.R]: I find the doc of this parameter is misning, compared with lgb.cv.
